ray_provider/operators/ray.py

```diff
@@ -175,6 +175,19 @@ def __init__(
         self.job_id = ""
         self.terminal_states = {JobStatus.SUCCEEDED, JobStatus.STOPPED, JobStatus.FAILED}
 
+        self.on_failure_callback = self._on_failure_callback
```

Comment:
I believe this approach may prevent users from utilizing `on_failure_callback` and `on_success_callback` for their specific use cases, which could lead to unexpected behaviour. For example, if a user wants to send a notification via email or Slack when a task succeeds or fails, they won’t be able to use both callbacks, as the current assignment would override them.

IMHO, we should either encourage the use of setup/teardown features for resource creation and cleanup, or manage resources based on a flag parameter without overriding these callbacks

Reply:
Removed both callbacks